uast/npath.go

```diff
@@ -0,0 +1,225 @@
+package uast
+
+import (
+	"errors"
+)
+
+//CodeReference
+//https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
+
+//NpathComplexity return a slice of int for each function in the tree
+func NpathComplexity(n *Node) ([]int, error) {
+	var funcs []*Node
+	var npath []int
+
+	if n.containsRol(FunctionDeclarationBody) {
+		funcs = append(funcs, n)
+	} else {
+		funcs = n.deepChildrenOfRole(FunctionDeclarationBody)
+	}
+	if len(funcs) <= 0 {
+		npath = append(npath, -1)
+		return npath, errors.New("Function declaration not found")
+	}
+	for _, function := range funcs {
+		npath = append(npath, visitFunctionBody(function))
+	}
+	return npath, nil
+}
+
+func visitorSelector(n *Node) int {
+	// I need to add a error when the node dont have any rol
+	// when I got 2 or more roles in a node this doesnt work
```

Comment:
Oh, I see. Do you think that's possible? The roles seem pretty different from each other...